inc/namespace.php

```diff
@@ -305,7 +305,12 @@ function send_trace_to_daemon( array $trace ) {
 function get_flattened_segments_from_trace( array $trace ) : array {
 	$max_size = 63 * 1024; // 63 KB, leaving room for UDP headers etc.
 	$segments = [];
-	if ( empty( $trace['subsegments'] ) || mb_strlen( json_encode( $trace ) ) < $max_size ) { // @codingStandardsIgnoreLine wp_json_encode not available.
+	$size = mb_strlen( json_encode( $trace ) ); // @codingStandardsIgnoreLine wp_json_encode not available.
+	if ( empty( $trace['subsegments'] ) || $size < $max_size ) {
+		// Truncate values until the trace is small enough to deliver.
+		if ( $size > $max_size ) {
+			$trace = truncate_trace( $trace );
```

Comment:
It looks like `truncate_trace` already does a check on the length, so we could avoid an additional `json_encode()` above by not checking it, and just always calling `truncase_trace()`?

Reply:
The size check there is the original one that determines whether to skip breaking a trace up into its subsegments. Removing it there would mean that every trace with subsegments will be sent individually which doesn't seem desirable?

We could pass the already known size to `truncate_trace()` as well, though makes it less of a unit.